Home/suggested-quiz.js

```diff
@@ -7,30 +7,96 @@ async function getImage(topic) {
     },
   });
   myImage = await image.json();
-  const newImage = document.createElement("img");
-  newImage.src = myImage.value[0].thumbnailUrl;
-  document.body.appendChild(newImage);
-  console.log(myImage);
+  actualImage = myImage.value[0].thumbnailUrl;
+  return actualImage;
 }
+
+const randomTopicsArray = [];
+const topic = {
+  easy: "General Knowledge",
+  medium: "General Knowledge",
+  hard: "General Knowledge",
+};
 // getImage("Entertainment");
-function playQuiz() {
-  console.log("Playing a quiz!");
-}
-document.querySelector(".suggested-quiz").addEventListener("click", playQuiz);
-function displaySuggestedQuizzes() {}
 
-function displayQuiz() {
+async function displayQuiz(difficulty, topic) {
   let quizzesSuggestionSection = document.querySelector(".suggested-quizzes");
   let quizDiv = document.createElement("div");
-  quizDiv.addEventListener("click", playQuiz);
   quizDiv.classList.add("suggested-quiz");
   quizzesSuggestionSection.appendChild(quizDiv);
 
   let quizImage = document.createElement("img");
-  quizImage.src = "testPhoto.jpg";
+  quizImage.src = await getImage(topic);
+  // quizImage.src = "testPhoto.jpg";
   quizDiv.appendChild(quizImage);
 
   const quizHeader = document.createElement("h2");
-  quizHeader.innerText = "Type of Quiz";
+  quizHeader.innerText = `${topic}: ${difficulty}`;
   quizDiv.appendChild(quizHeader);
+
+  return quizDiv;
+}
+function generateRandomIdArray() {
+  while (randomTopicsArray.length < 3) {
+    let a = Math.floor(Math.random() * categoryArray.length);
+    if (randomTopicsArray.indexOf(a) === -1) randomTopicsArray.push(categoryArray[a]);
+  }
+}
+async function getQuizTopics(id, difficulty) {
+  let responce = await fetch(`https://opentdb.com/api.php?amount=10&category=${id}`);
+  let quizJSON = await responce.json();
+  let quiz = quizJSON.results;
+  topic[difficulty] = quiz[0].category;
+  let quizName = quiz[0].category;
+  if (quizName.search("Entertainment:") !== -1) {
+    quizName = quizName.slice(14);
+  }
+  if (quizName.search("Science:") !== -1) {
+    quizName = quizName.slice(8);
+  }
+  return quizName;
+}
+
+async function generateEasyQuiz() {
+  const easyQuizName = await getQuizTopics(randomTopicsArray[0], "easy");
+  const easyDiv = await displayQuiz("Easy", easyQuizName);
+  easyDiv.addEventListener("click", playEasyQuiz);
+}
+async function generateMediumQuiz() {
+  const MediumQuizName = await getQuizTopics(randomTopicsArray[1], "medium");
```

Comment:
Maybe just avoid Pascal casing here and switch to camel casing instead ~

Reply:
Not a biggie but just so that I can play with github I am going to ask you to change this via the review changes just to see how ti works! Hope you dont mind ~

Reply:
Good spot!